lib/renderer/web-view/web-view-impl.ts

```diff
@@ -194,11 +200,16 @@ export class WebViewImpl {
     this.internalInstanceId = getNextId();
     this.guestInstanceId = guestInstanceId;
 
-    guestViewInternal.attachGuest(
+    const embedderFrameId = this.hooks.getWebFrameId(this.internalElement.contentWindow!);
```

Comment:
Can this potentially be spoofed? Can the main world content fake the value of `this.internalElement.contentWindow` and get access to a frame it wouldn't normally have access to?

Reply:
I would be more worried about IPC argument validation on the main process side.

Reply:
We end up calling this https://github.com/electron/electron/blob/e01faedaa5ec6e391e1290d7c29231e0811a32e2/shell/browser/web_view_guest_delegate.cc#L29-L52

The `DCHECK_EQ` checks if the given frame belongs to the embedder webContents. This should probably fail instead to prevent the guest webContents being attached to a belonging to a different parent webContents, which happens to live in the same renderer process.

Reply:
@nornagon what do you think about ^^^? I would address that in a separate PR, which should be backported

Reply:
Yeah, I like the idea of validating this on the browser side.